velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -99,12 +99,53 @@ FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::BOOLEAN>(
   });
 }
 
+template <>
+FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::BIGINT>(
+    const SelectivityVector& rows,
+    BufferPtr& hashes) {
+  if (vector_->base()->type()->isShortDecimal()) {
+    applyHashFunction(rows, *vector_.get(), hashes, [&](auto row) {
+      return vector_->valueAt<int64_t>(row);
+    });
+  } else {
+    applyHashFunction(rows, *vector_.get(), hashes, [&](auto row) {
+      return hashInteger(vector_->valueAt<int64_t>(row));
+    });
+  }
+}
+
+FOLLY_ALWAYS_INLINE uint64_t mix(uint64_t current, uint64_t value) {
+  return XXH_rotl64(current + value * XXH_PRIME64_2, 31) * XXH_PRIME64_1;
+}
+
+FOLLY_ALWAYS_INLINE uint64_t updateTail(uint64_t hash, uint64_t value) {
+  auto temp = hash ^ mix(0, value);
+  return XXH_rotl64(temp, 27) * XXH_PRIME64_1 + XXH_PRIME64_4;
+}
+
+FOLLY_ALWAYS_INLINE uint64_t hashLongDecimalPart(const uint64_t value) {
+  auto hash = XXH_PRIME64_5 + sizeof(uint64_t);
+  hash = updateTail(hash, value);
+  hash = XXH64_avalanche(hash);
+  return hash;
+}
+
 template <>
 FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::HUGEINT>(
     const SelectivityVector& rows,
     BufferPtr& hashes) {
+  constexpr int64_t kInt64Mask = ~(static_cast<int64_t>(1) << 63);
```

Comment:
@wypb thanks for the background on the hashing. Can you add that link in a comment?
We also need to add a comment that the computation here is to translate the 2's complement C++ value to Java's signed magnitude representation.
There is a slightly different implementation in `PrestoSerialization.cpp:readJavaDecimal()`.
Should we add this conversion to `DecimalUtil.h`?

Reply:
@majetideepak I'll add some comments here, In addition, this place actually refers to `PrestoSerialization.cpp::toJavaDecimalValue` to translate the 2's complement C++ value to Java's signed magnitude representation, mainly for the convenience of hash computation.